api/src/controllers/provision.ts

```diff
@@ -32,23 +32,35 @@ import shared from '../libs/shared';
 const dm = new DataManager(shared.pgPool);
 
 export const provisionProfileNamespaces = async (
-  { params, user }: { params: any, user: AuthenticatedUser }, res: Response
+  { params, body, user }: { params: any, body: any, user: AuthenticatedUser }, res: Response
 ): Promise<void> => {
   const { profileId } = params;
+  const clusters = body;
   const { ProfileModel, NamespaceModel, ClusterModel } = dm;
 
   try {
     const existing = await NamespaceModel.findForProfile(profileId);
     if (existing.length === 0) {
       const profile = await ProfileModel.findById(profileId);
-      const cluster = await ClusterModel.findByName(profile.primaryClusterName);
 
-      if (!profile || !cluster) {
+      if (!profile) {
         const errmsg = 'Unable to fetch info for provisioning';
         throw new Error(errmsg);
       }
 
-      await NamespaceModel.createProjectSet(profileId, Number(cluster.id), profile.namespacePrefix);
+      const names = projectSetNames.map(n => `${profile.namespacePrefix}-${n}`);
```

Comment:
Consider refactoring this block for readability. It may read better top to bottom of everything was a for..of. Admittedly I'm a bit on the fence but I find dealing mixing strategies for dealing with await a bit harder to read.